commit 184af728c73c379d1eee57a387b6012572794fa8
Author: Kevin Easton <caf@bitchx.org>
Date:   Fri Mar 17 00:38:46 2017 +1100

    Add support for OpenSSL 1.1.0 to configure script
    
    OpenSSL 1.1.0 removed the SSLeay() function and replaced it with a macro, that AC_CHECK_LIB can't see.
    
    Search for ERR_get_error() instead.

diff --git configure configure
index d435491..f4af7bb 100755
--- configure
+++ configure
@@ -10630,9 +10630,9 @@ esac
 case "$with_ssl" in
     yes|check)
 
-echo "$as_me:$LINENO: checking for SSLeay in -lcrypto" >&5
-echo $ECHO_N "checking for SSLeay in -lcrypto... $ECHO_C" >&6
-if test "${ac_cv_lib_crypto_SSLeay+set}" = set; then
+echo "$as_me:$LINENO: checking for ERR_get_error in -lcrypto" >&5
+echo $ECHO_N "checking for ERR_get_error in -lcrypto... $ECHO_C" >&6
+if test "${ac_cv_lib_crypto_ERR_get_error+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -10650,11 +10650,11 @@ extern "C"
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char SSLeay ();
+char ERR_get_error ();
 int
 main ()
 {
-SSLeay ();
+ERR_get_error ();
   ;
   return 0;
 }
@@ -10681,20 +10681,20 @@ if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_crypto_SSLeay=yes
+  ac_cv_lib_crypto_ERR_get_error=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_crypto_SSLeay=no
+ac_cv_lib_crypto_ERR_get_error=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_SSLeay" >&5
-echo "${ECHO_T}$ac_cv_lib_crypto_SSLeay" >&6
-if test $ac_cv_lib_crypto_SSLeay = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_ERR_get_error" >&5
+echo "${ECHO_T}$ac_cv_lib_crypto_ERR_get_error" >&6
+if test $ac_cv_lib_crypto_ERR_get_error = yes; then
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBCRYPTO 1
 _ACEOF
@@ -10703,7 +10703,7 @@ _ACEOF
 
 fi
 
-      if test x"$ac_cv_lib_crypto_SSLeay" = x"yes"; then
+      if test x"$ac_cv_lib_crypto_ERR_get_error" = x"yes"; then
 
 echo "$as_me:$LINENO: checking for SSL_accept in -lssl" >&5
 echo $ECHO_N "checking for SSL_accept in -lssl... $ECHO_C" >&6
diff --git configure.in configure.in
index 63d133f..3694a7f 100644
--- configure.in
+++ configure.in
@@ -562,8 +562,8 @@ case "$with_ssl" in
 esac
 case "$with_ssl" in
     yes|check)
-      AC_CHECK_LIB([crypto], [SSLeay], [], [], [])
-      if test x"$ac_cv_lib_crypto_SSLeay" = x"yes"; then
+      AC_CHECK_LIB([crypto], [ERR_get_error], [], [], [])
+      if test x"$ac_cv_lib_crypto_ERR_get_error" = x"yes"; then
         AC_CHECK_LIB([ssl], [SSL_accept], [], [], [])
       fi
       if test x"$ac_cv_lib_ssl_SSL_accept" = x"yes"; then
